R/extract.R

```diff
@@ -0,0 +1,174 @@
+#' Extract elements of a workflow
+#'
+#' @description
+#' These functions extract various elements from a workflow object. If they do
+#' not exist yet, an error is thrown.
+#'
+#' - `extract_preprocessor()` returns the formula, recipe, or variable
+#'   expressions used for preprocessing.
+#'
+#' - `extract_spec_parsnip()` returns the parsnip model specification.
+#'
+#' - `extract_fit_parsnip()` returns the parsnip model fit object.
+#'
+#' - `extract_fit_engine()` returns the engine specific fit embedded within
+#'   a parsnip model fit. For example, when using [parsnip::linear_reg()]
+#'   with the `"lm"` engine, this would return the underlying `lm` object.
+#'
+#' - `extract_mold()` returns the preprocessed "mold" object returned
+#'   from [hardhat::mold()]. It contains information about the preprocessing,
+#'   including either the prepped recipe, the formula terms object, or
+#'   variable selectors.
+#'
+#' - `extract_recipe()` returns the recipe. The `estimated` argument specifies
+#'    whether the fitted or original recipe is returned.
```

Comment:
since the parameter is called `estimated`, do we want to also consistently talk about estimated recipes instead of fitted recipes? For workflows, workflowsets, and tune, this pops up here and in the `@param` documentation. For workflows and tune, this also shows up in the comments of the examples.

Reply:
I'm good with "estimated" and "fitted"; I'm trying to get away from "prepared"